### PR TITLE
Add support for HJSON and YAML

### PR DESCRIPTION
**Pull request by davehunt**
This builds on the work from pull #6. @the-ivan would you mind taking a look over my additions?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-  - pypy
-  - pypy3
 env:
-  - PYTEST=2.6.4
-  - PYTEST=2.7.2
+  - TOXENV=py26
+  - TOXENV=py27
+  - TOXENV=py32
+  - TOXENV=py33
+  - TOXENV=py34
```
Comment (the-ivan) [inline .travis.yml]: I am wondering if there should be an extra entry for py35?

File: test_variables.py [hunk 2]
```diff
@@ -2,66 +2,103 @@
...
+        metafunc.parametrize('file_format', ['json', 'hjson', 'yaml'])
+
+
+def run(testdir, file_format='json', variables=None):
+    variables = variables or [{"foo": "bar"}]
     args = []
     for i, v in enumerate(variables):
+        if file_format == 'hjson':
+            if (3, 0) <= sys.version_info < (3, 3):
+                pytest.skip('hjson oes not support Python < 3.3')
```
Comment (the-ivan) [inline test_variables.py]: not: small typo in pytest.skip()

File: .travis.yml (hunk 1 above)
Comment (davehunt) [inline .travis.yml]: There should, but the way this works it depends on the default Python image in Travis having all of these versions available. At the moment, Python 3.5 is not available. See https://github.com/travis-ci/travis-ci/issues/4794. There is a workaround it seems, or I could make this expected to fail for now.

**Comment by the-ivan** [comment]
Dave, other than the two comments, LGTM. Thank you for making my first pull request a fantastic learning experience!

**Comment by davehunt** [comment]
Thanks for the review @the-ivan, I've added Python 3.5 to Travis now and found a better way to skip formats on particular Python versions (PyYAML fails in 3.5). I'll go ahead and merge/release this now. Thank you for your contribution!

**Comment by davehunt** [comment]
I am curious, @the-ivan, how did you find out about this plugin, and what made you want to contribute? Are you using it in your test suites?

**Comment by the-ivan** [comment]
Thanks @davehunt! I started a new job in Seattle and as they had absolutely no automation here, I took on the role of implementing a solution. I've worked with Webdriver in Ruby before, but wanted to use Python this time around. I found pytest but wanted something more specific to Selenium, then somehow stumbled across Mozilla's mozwebqa examples which in turn led me to pytest-selenium (making everything so much easier). It was only much later than I realized that you are the mastermind behind both!

We're not using POM here as the learning curve for OO is a bit too high for the manual testers, so I wanted a central repository for our variables for easier maintenance. Commenting is an obvious feature, thus that original pull request. I'm slightly amazed at the difference between that and the code merged today. Thanks again for all the guidance!

I have a lot on my plate currently but would like to work on the other two plugins, pytest-selenium and pytest-html, as well. Looking forward to further collaboration!

**Comment by davehunt** [comment]
That's great - I love Seattle! :smile: You may also be interested in [PyPOM](https://github.com/davehunt/PyPOM), which is an abstraction (and slight evolution) of our page object model. It's still a work in progress, but hopefully we'll have a version released in the next couple of months. Thanks again for the patch!

**Pull request merged**